main.js

```diff
@@ -0,0 +1,181 @@
+/*!
+GPII Electron
+Copyright 2016 Steven Githens
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+The research leading to these results has received funding from the European Union's
+Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+"use strict";
+
+var fluid = require("infusion");
+var gpii = fluid.registerNamespace("gpii");
+const {app, Menu, Tray} = require('electron')
+var path = require("path");
+var request = require("request");
+var os = require("os");
+
+// The tray needs to be global
+var tray = null;
+
+// Component to start and stop parts of the GPII as well as key in and key out users
+fluid.defaults("gpii.app", {
+    gradeNames: "fluid.modelComponent",
+    model: {
+        gpiiStarted: false,
+        keyedInSet: null
+    },
+    listeners: {
+        onCreate: [{
+            funcName: "gpii.app.startLocalFlowManager"
+        }, {
+            funcName: "{that}.changeStarted",
+            args: [true]
+        }]
+    },
+    modelListeners: {
+        keyedInSet: [{ //  Will this be a race condition?
```

Comment:
Yes, it probably will. Please put priority-based constraints on these if you want them to execute in a particular order:

http://docs.fluidproject.org/infusion/development/Priorities.html#table-of-supported-sites-for-priorities-and-namespaces